cvat/apps/engine/static/engine/js/annotationSaver.js

```diff
@@ -241,20 +245,33 @@ class AnnotationSaverModel extends Listener {
                 this._shapeCollection.flush = false;
                 this._version = savedObjects.version;
                 this._resetState();
+
                 for (const type of Object.keys(this._initialObjects)) {
                     for (const object of savedObjects[type]) {
+                        if (object.shapes) {
+                            for (const shape of object.shapes) {
+                                delete shape.id;
+                            }
+                        }
                         this._initialObjects[type][object.id] = object;
                     }
                 }
+
                 this._version = savedObjects.version;
             } else {
                 const [created, updated, deleted] = this._split(exported);
                 this.notify('saveCreated');
                 const savedCreated = await this._create(created);
                 this._updateCreatedObjects(created, savedCreated, mapping);
                 this._version = savedCreated.version;
+
                 for (const type of Object.keys(this._initialObjects)) {
                     for (const object of savedCreated[type]) {
+                        if (object.shapes) {
```

Comment:
Does it make sense to create a special function here and use it 3 times?

Reply:
Probably, it does

Reply:
I missed that it is from legacy UI. Let's keep it as is for now.